confidence-core/src/main/java/org/saynotobugs/confidence/quality/function/DelegateMapsArgument.java

```diff
@@ -0,0 +1,77 @@
+package org.saynotobugs.confidence.quality.function;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import org.dmfs.srcless.annotations.staticfactory.StaticFactories;
+import org.saynotobugs.confidence.Assessment;
+import org.saynotobugs.confidence.Description;
+import org.saynotobugs.confidence.Quality;
+import org.saynotobugs.confidence.assessment.FailUpdated;
+import org.saynotobugs.confidence.description.Spaced;
+import org.saynotobugs.confidence.description.Text;
+import org.saynotobugs.confidence.description.Value;
+
+/**
+ * This quality can be used to isolate the logic of the outer function in a delegating function,
+ * which enables testing only the actual functions logic, not the logic of the delegate function.
+ *
+ * <pre>
+ * OuterFunction
+ * +-----------------------------------------------------------------------------------+
+ * |                      DelegateFunction                                             |
+ * |                     +------------------------------------------+                  |
+ * | OuterArgument ----> |  DelegateArgument -----> DelegateResult  | ---> OuterResult |
+ * |                     +------------------------------------------+                  |
+ * +-----------------------------------------------------------------------------------+
+ * </pre>
+ */
+@StaticFactories(value = "Core", packageName = "org.saynotobugs.confidence.quality")
+public final class DelegateMapsArgument<In, DelegateIn, Out> implements Quality<Function<Function<DelegateIn, Out>, Function<In, Out>>> {
+	private final In mOuterArgument;
+	private final Out mDelegateResult;
+	private final Quality<DelegateIn> mDelegateArgumentQuality;
+
+	/**
+	 * Creates new instance.
+	 *
+	 * @param outerArgument  outer argument
+	 * @param delegateResult delegate result, no impact on quality, but must be a valid result that does not cause an excaption
+	 * @param delegateArgumentQuality quality verifying the delegate argument
+	 */
+	public DelegateMapsArgument(In outerArgument, Out delegateResult, Quality<DelegateIn> delegateArgumentQuality) {
```

Comment:
In general a delegate `Quality`s should always be contravariant, i.e. passed as `Quality<? super DelegateIn>`